pkg/apiserver/registry/core/namespacedcloudprofile/strategy_test.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2024 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package namespacedcloudprofile_test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/gardener/gardener/pkg/apis/core"
+	namespacedcloudprofileregistry "github.com/gardener/gardener/pkg/apiserver/registry/core/namespacedcloudprofile"
+)
+
+var _ = Describe("PrepareForCreate", func() {
+	var namespacedCloudProfile *core.NamespacedCloudProfile
+
+	It("should drop the expired Kubernetes and MachineImage versions from the namespacedcloudprofile", func() {
```

Comment:
Please add another test where Kubernetes is not set, but machine images and the other way around.

Reply:
Improved the test in https://github.com/gardener/gardener/pull/9440/commits/0fd4869199523426d04513d4b4346724314e7bff